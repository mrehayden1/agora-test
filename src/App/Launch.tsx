import { useState } from 'react';

export type LaunchData = {
  details: string;
  launch_date_unix: number;
  mission_name: string;
  rocket: {
    rocket_name: string;
    rocket_type: string;
  };
};

export default function Launch(props: LaunchData) {

  const [ open, setOpen ] = useState(false);

  return (
    <div>
      <h2>{props.mission_name}</h2>
      <dl>
        <dt>Date</dt>
        <dd>{(new Date(props.launch_date_unix * 1000)).toLocaleString()}</dd>
        <dt>Launch details</dt>
        <dd>{props.details || '(no details)'}</dd>
      </dl>
      {!open ? (
        <button onClick={() => setOpen(true)}>view rocket details</button>
      ) : (
        <>
          <h3 className="sr-only">Rocket details</h3>
          <dl>
            <dt>Rocket name</dt>
            <dd>{props.rocket.rocket_name}</dd>
            <dt>Rocket type</dt>
            <dd>{props.rocket.rocket_type}</dd>
          </dl>
          <button onClick={() => setOpen(false)}>hide rocket details</button>
        </>
      )}
    </div>
  );

}
