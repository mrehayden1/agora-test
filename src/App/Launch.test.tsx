import { shallow } from 'enzyme';
import React from 'react';

import Launch from './Launch';

test('renders matching snapshot', () => {
  const launch = {
    details: 'details',
    launch_date_unix: 1631655161,
    mission_name: 'mission name',
    rocket: {
      rocket_name: 'rocket name',
      rocket_type: 'rocket type'
    }
  };

  const component = shallow(<Launch {...launch} />);
  expect(component).toMatchSnapshot();
});

test('renders matching snapshot after expanding rocket details', () => {
  const launch = {
    details: 'details',
    launch_date_unix: 1631655161,
    mission_name: 'mission name',
    rocket: {
      rocket_name: 'rocket name',
      rocket_type: 'rocket type'
    }
  };

  const component = shallow(<Launch {...launch} />);
  component.find('button').simulate('click');
  expect(component).toMatchSnapshot();
});
