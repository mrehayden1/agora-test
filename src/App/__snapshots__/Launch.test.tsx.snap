// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders matching snapshot 1`] = `
<div>
  <h2>
    mission name
  </h2>
  <dl>
    <dt>
      Date
    </dt>
    <dd>
      9/14/2021, 22:32:41
    </dd>
    <dt>
      Launch details
    </dt>
    <dd>
      details
    </dd>
  </dl>
  <button
    onClick={[Function]}
  >
    view rocket details
  </button>
</div>
`;

exports[`renders matching snapshot after expanding rocket details 1`] = `
<div>
  <h2>
    mission name
  </h2>
  <dl>
    <dt>
      Date
    </dt>
    <dd>
      9/14/2021, 22:32:41
    </dd>
    <dt>
      Launch details
    </dt>
    <dd>
      details
    </dd>
  </dl>
  <h3
    className="sr-only"
  >
    Rocket details
  </h3>
  <dl>
    <dt>
      Rocket name
    </dt>
    <dd>
      rocket name
    </dd>
    <dt>
      Rocket type
    </dt>
    <dd>
      rocket type
    </dd>
  </dl>
  <button
    onClick={[Function]}
  >
    hide rocket details
  </button>
</div>
`;
